### ISSUE TITLE
Rule proposal: Prefer declaring variables together only when they are using destructuring of the same variable

### ISSUE THREAD

**Issue by wenfangdu** [sindresorhus/eslint-plugin-unicorn#2092]
### Description

Current:

```js
const foo = {}
const { bar } = foo
const { baz } = foo
```

Desired:

```js
const foo = {}
const { bar, baz } = foo
```

### Fail

```js
const foo = {}
const { bar } = foo
const { baz } = foo
```

### Pass

```js
const foo = {}
const { bar, baz } = foo
```

**Comment by fregante** [comment]
Related:

- https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/consistent-destructuring.md
